scripts/mkcloud

```diff
@@ -271,213 +275,52 @@ function onhost()
 
 function cleanup()
 {
-    # cleanup leftover from last run
-    ${mkcloud_lib_dir}/libvirt/cleanup $cloud $nodenumber $cloudbr $vlan_public
-
-    if ip link show ${cloudbr}.$vlan_public >/dev/null 2>&1; then
-        ip link set ${cloudbr}.$vlan_public down
-    fi
-    if ip link show ${cloudbr} >/dev/null 2>&1; then
-        ip link set ${cloudbr} down
-        ip link delete ${cloudbr} type bridge
-        ip link delete ${cloudbr}-nic
-    fi
-    # 1. remove leftover partition mappings that are still open for this cloud
-    local vol
-    dmsetup ls | awk "/^$cloudvg-${cloud}\./ {print \$1}" | while read vol ; do
-        kpartx -dsv /dev/mapper/$vol
-    done
-
-    # workaround host grabbing guest devices
-    for vol in postgresql rabbitmq ; do
-        dmsetup remove drbd-$vol
-    done
-    # 2. remove all previous volumes for that cloud; this helps preventing
-    # accidental booting and freeing space
-    if [ -d $vdisk_dir ]; then
-        find -L $vdisk_dir -name "$cloud.*" -type b | \
-            xargs --no-run-if-empty lvremove --force || complain 104 "lvremove failure"
-    fi
-    rm -f /etc/lvm/archive/*
-
-    if [[ $wipe = 1 ]] ; then
-        vgchange -an $cloudvg
-        dd if=/dev/zero of=$cloudpv count=1000
-    fi
-    return 0
+    do_cleanup
 }
 
-
-function onhost_cleanup_admin_node()
+function do_cleanup()
 {
-    # this function is meant to only clean the admin node
-    # in order to deploy a new one, while keeping all cloud nodes
-
-    ${mkcloud_lib_dir}/libvirt/cleanup_one_node ${cloud}-admin
+    ${mkclouddriver}_do_cleanup
 }
 
-function onhost_get_next_pv_device()
+function onhost_cleanup_admin_node()
 {
-    if [ -z "$pvlist" ] ; then
-        pvlist=`pvs --sort -Free | awk '$2~/'$cloudvg'/{print $1}'`
-        pv_cur_device_no=0
-    fi
-    next_pv_device=`perl -e '$i=shift; $i=$i % @ARGV;  print $ARGV[$i]' $pv_cur_device_no $pvlist`
-    pv_cur_device_no=$(( $pv_cur_device_no + 1 ))
+    ${mkclouddriver}_do_cleanup_admin_node
 }
 
-# create lv device wrapper
-function _lvcreate()
+function onhost_get_next_pv_device()
 {
-    lv_name=$1
-    lv_size=$2
-    lv_vg=$3
-    lv_pv=$4
-
-    # first: create on the PV device (spread IO)
-    # fallback: create in VG (if PVs with different size exist)
-    lvcreate -n $lv_name -L ${lv_size}G $lv_vg $lv_pv || \
-        safely lvcreate -n $lv_name -L ${lv_size}G $lv_vg
+    ${mkclouddriver}_do_get_next_pv_device
 }
 
-# spread block devices over a LVM's PVs so that different VMs
-# are likely to use different PVs to optimize concurrent IO throughput
 function onhost_create_cloud_lvm()
```

Comment:
see the discussion with jdsn, he wants to have the onhost_ functions to stay around. I am not sure we want them to be moved to lib/mkcloud-libvirt.sh as that would mean that they would get the libvirt_ prefix which makes them loose the onhost prefix..